app/src/main/java/org/scesi/cappuchino/ui/screens/homeScreen.kt

```diff
@@ -1,64 +1,71 @@
 package org.scesi.cappuchino.ui.screens
 
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.expandVertically
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.shrinkVertically
+import androidx.compose.animation.slideInVertically
+import androidx.compose.animation.slideOutVertically
 import androidx.compose.foundation.Image
-import androidx.compose.foundation.background
-import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.Row
 import androidx.compose.foundation.layout.Spacer
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.layout.height
 import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.layout.size
 import androidx.compose.foundation.layout.wrapContentHeight
-import androidx.compose.foundation.rememberScrollState
-import androidx.compose.foundation.shape.RoundedCornerShape
-import androidx.compose.foundation.verticalScroll
-import androidx.compose.material3.ExperimentalMaterial3Api
-import androidx.compose.material3.Icon
-import androidx.compose.material3.OutlinedTextField
 import androidx.compose.material3.Text
-import androidx.compose.material3.TextFieldDefaults
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.draw.clip
-import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.dimensionResource
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextAlign
-import androidx.compose.ui.unit.sp
-import org.scesi.cappuchino.R
-import androidx.compose.ui.res.dimensionResource
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.scesi.cappuchino.R
 import org.scesi.cappuchino.ui.theme.CappuchinoTheme
+import org.scesi.cappuchino.ui.utils.SearchBar
+import org.scesi.cappuchino.ui.utils.SearchCategory
 
 
 @Composable
 fun HomeScreen() {
-    val title = stringResource(id = R.string.titulo_main) // Obtiene la cadena desde strings.xml
+    val title = stringResource(id = R.string.titulo_main)
+    var searchBarVisible by remember { mutableStateOf(true) }
     CappuchinoScaffold(title) {
-        HomeScreenContent(title)
+        HomeScreenContent(
+            searchBarVisible = searchBarVisible,
+            onSearchBarClicked = { searchBarVisible = false }
```

Comment:
take into consideration that this searchBarVisible is for the logo and the welcome message , not for the search bar itself